jobs/async-upload/tests/integration/test_integration_async_upload.py

```diff
@@ -0,0 +1,378 @@
+    final_key = path_parts[-1]
+    
+    # Check if final key is a numeric index
+    if final_key.isdigit():
+        index = int(final_key)
+        current[index] = value
+    else:
+        current[final_key] = value
+
+
+def apply_resource(resource: Dict[str, Any], k8s_client) -> None:
+    """Apply a single Kubernetes resource."""
+    try:
+        from kubernetes.dynamic import DynamicClient
+    except ImportError:
+        pytest.skip("kubernetes.dynamic not available")
+    
+    dyn_client = DynamicClient(k8s_client)
+    
+    # Get the API version and kind
+    api_version = resource.get("apiVersion", "v1")
+    kind = resource.get("kind")
+    name = resource["metadata"]["name"]
+    namespace = resource["metadata"].get("namespace", "default")
+    
+    # Create the resource
+    api_resource = dyn_client.resources.get(api_version=api_version, kind=kind)
+    
+    # For Jobs, delete existing one first to avoid conflicts
+    if kind == "Job":
+        try:
+            api_resource.delete(name=name, namespace=namespace)
+            time.sleep(2)  # Wait for deletion to complete
+        except Exception:
+            pass  # Ignore errors if job doesn't exist
+    
+    # Apply the resource
+    try:
+        api_resource.create(body=resource)
+    except Exception as e:
+        if hasattr(e, 'status') and getattr(e, 'status', None) == 409:  # Conflict - resource already exists
+            # For non-Job resources, just ignore if they already exist
+            if kind != "Job":
+                print(f"Resource {kind}/{name} already exists, skipping...")
+                return
+            else:
+                raise  # Jobs should have been deleted above, so this is unexpected
+        else:
+            raise
+
+
+def wait_for_job_completion(
+    job_name: str,
+    namespace: str,
+    batch_client,
+    timeout_seconds: int = 600
+) -> bool:
+    """Wait for job completion and return success status."""
+    start_time = time.time()
+    
+    while time.time() - start_time < timeout_seconds:
+        try:
+            job = batch_client.read_namespaced_job(name=job_name, namespace=namespace)
+            
+            # Check if job is complete
+            if job.status.conditions:
+                for condition in job.status.conditions:
+                    if condition.type == "Complete" and condition.status == "True":
+                        return True
+                    elif condition.type == "Failed" and condition.status == "True":
+                        return False
+            
+            time.sleep(10)  # Wait 10 seconds before checking again
+            
+        except Exception as e:
+            if hasattr(e, 'status') and getattr(e, 'status', None) == 404:
+                # Job doesn't exist yet
+                time.sleep(5)
+                continue
+            else:
+                raise
+    
+    return False
+
+
+def upload_to_minio(file_path: str, bucket: str, key: str) -> None:
```

Comment:
this works for me, but since boto3 is already part of our dev/test dependencies, wouldn't be easier to just use boto3 here?